### ISSUE TITLE
[RFC] resources tagged by cloudformation

### ISSUE THREAD

**Issue by endemics** [dtan4/terraforming#58]
Here's my problem: I'm trying to dump security groups from an environment containing resources managed by CloudFormation.

Cloudformation tags resources it manages such as SecurityGroups with tags following the pattern `aws:cloudformation:*`. However this pattern is not supported by Hashicorp's hcl lexer as ':' is not in the list of supported tokens in https://github.com/hashicorp/hcl/blob/master/hcl/lex.go#L199

At this point, I see 3 alternatives:
- try to get hashicorp to support ':' (very unlikely as it will probably have a lot of side effects for them)
- quote the tags with something like:
```diff
--- a/lib/terraforming/template/tf/security_group.erb
+++ b/lib/terraforming/template/tf/security_group.erb
@@ -36,7 +36,7 @@ resource "aws_security_group" "<%= module_name_of(security_group) %>" {
 <% if security_group.tags.length > 0 -%>
     tags {
 <% security_group.tags.each do |tag| -%>
-        <%= tag.key %> = "<%= tag.value %>"
+        "<%= tag.key %>" = "<%= tag.value %>"
 <% end -%>
     }
 <% end -%>
```
- completely ignore (not dump) resources managed by CloudFormation when we can detect them.

**Comment by dtan4** [comment]
Thank you for reporting issue :octocat: 

The document [Tagging Your Amazon EC2 Resources - Amazon Elastic Compute Cloud](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html) says:

> Generally allowed characters are: letters, spaces, and numbers representable in UTF-8, plus the following special characters: + - = . _ : / @.

So we sometimes want to use ':' as tag name, which is not allowed to HCL parser.

It seems to be good to quote the key.
Quoted name is valid in Terraform.

```
resource "aws_security_group" "hoge" {
  tags {
    "hoge:fuga" = "fuga"
  }
}
```

```bash
$ terraform plan
The Terraform execution plan has been generated and is shown below.
Resources are shown in alphabetical order for quick scanning. Green resources
will be created (or destroyed and then created if an existing resource
exists), yellow resources are being changed in-place, and red resources
will be destroyed.

Note: You didn't specify an "-out" parameter to save this plan, so when
"apply" is called, Terraform can't guarantee this is what will execute.

+ aws_security_group.hoge
    description:    "" => "Managed by Terraform"
    egress.#:       "" => "<computed>"
    ingress.#:      "" => "<computed>"
    name:           "" => "<computed>"
    owner_id:       "" => "<computed>"
    tags.#:         "" => "1"
    tags.hoge:fuga: "" => "fuga"
    vpc_id:         "" => "<computed>"
```

I will support them ASAP.

**Comment by dtan4** [comment]
@endemics I implemented quoted tag name at #59 and released it as v0.0.5.
Please check it :smile:

**Comment by endemics** [comment]
Thanks @dtan4 I confirm that terraform now parses without error a tf file generated with terraforming that contains cloudformation-generated tags.
